src/lib.rs

```diff
@@ -1,3 +1,117 @@
+//!    typesetting. The wort character is often used as a lay term for grapheme, and one could
+//!    argue that given their relative usage that this is character's canonical meaning. In the `c`
+//!    programming language, a `char` is almost always just a name for a byte of data; this meaing
+//!    comes from a time when computer software engineers only considered English, and so a byte
+//!    was enough space to encode all possible glyphs.  Within rust, characters are specific
+//!    unicode code points, which are combined into graphemes. They don't really have any semantic
+//!    meaning on their own, although many characters consist of 1 grapheme, and so they are
+//!    sometimes mistakenly thought of as equivalent.
+//!  - *unicode* - When computers were first created in America, they all used the latin alphabet,
+//!    which fits into a byte. The *ASCII* encoding was developed to store latin characters in a
+//!    byte of data. As other countries began to use computers, they created their own encodings to
+//!    include their non-latin letters (for example É in French). Asian languages had to widen the
+//!    character to 2 or more bytes since there are more than 256 glyphs in these languages.
+//!    Unicode was developed as a (successful as it turns out) attempt to unify all languages and
+//!    scripts into a single encoding scheme. Unicode characters are 4 bytes wide, meaning that
+//!    there is space for 4_294_967_296 different characters, which is expected to be enough. (It's
+//!    not quite this many in practice - some bit patterns are reserved).
+//!  - *utf-8* - Unicode was a very useful in making it easier to use the same software in
+//!    different parts of the world, but if all your text is latin then it takes 4 bytes per
+//!    character instead of 1 for ASCII. This would be quite wasteful, so the most popular way to
+//!    store unicode strings (utf-8) uses a continuation-bit style scheme to compress unicode
+//!    characters where possible. Another older scheme is utf-16, which works similarly, but uses 2
+//!    bytes (16 bits) as its smallest size.
+//!  - *grapheme, grapheme cluster* - Outside of computer typesetting, graphemes can have multiple
+//!    meanings in linguistics (see wikipedia). Within computer typesetting (and unicode in
+//!    particular), a grapheme cluster (called grapheme in the rest of these docs) can be thought
+//!    of as the conceptual version of a glyph. One exception to this is Han unification
+//!    (see below). It's important to note that a grapheme cluster may be 1 or more unicode
+//!    characters.  An example of a multi-character grapheme cluster is a rainbow flag (`U+1F3F3
+//!    (white flag), U+200D (zwj), U+1F308 (rainbow)`) which produces a single rainbow flag glyph.
+//!  - *script* - A set of graphemes that are used together to achieve written communcation.
+//!  - *language* - A context for deriving meaning from a script. Collections of graphemes from a
```

Comment:
You might want to introduce the term "writing system" here, which is similar to script but subtly different. See https://en.wikipedia.org/wiki/Script_(Unicode) and https://en.wikipedia.org/wiki/Writing_system

Reply:
I've replaced the script section with a TODO.